packages/run-protocol/src/vpool-xyk-amm/pool.js

```diff
@@ -43,210 +43,257 @@ export const makeAddPool = (
+      state.liqTokenSupply,
+      zcfSeat.getStagedAllocation().Central.value,
+      poolCentralAmount.value,
+    );
 
-        // Central was specified precisely so offer must provide enough secondary.
-        assert(
-          AmountMath.isGTE(secondaryIn, secondaryRequired),
-          'insufficient Secondary deposited',
-        );
+    const liquidityAmountOut = AmountMath.make(
+      liquidityBrand,
+      liquidityValueOut,
+    );
+    liquidityZcfMint.mintGains(
+      harden({ Liquidity: liquidityAmountOut }),
+      poolSeat,
+    );
+    state.liqTokenSupply += liquidityValueOut;
+
+    poolSeat.incrementBy(
+      zcfSeat.decrementBy(
+        harden({
+          Central: zcfSeat.getStagedAllocation().Central,
+          Secondary: secondaryAmount,
+        }),
+      ),
+    );
 
-        return addLiquidityActual(pool, zcfSeat, secondaryRequired, centralIn);
-      },
-      removeLiquidity: userSeat => {
-        const liquidityIn = userSeat.getAmountAllocated(
-          'Liquidity',
-          liquidityBrand,
-        );
-        const liquidityValueIn = liquidityIn.value;
-        assert(isNatValue(liquidityValueIn), 'User Liquidity');
-        const centralTokenAmountOut = AmountMath.make(
-          centralBrand,
-          calcValueToRemove(
-            liqTokenSupply,
-            pool.getCentralAmount().value,
-            liquidityValueIn,
-          ),
-        );
+    zcfSeat.incrementBy(
+      poolSeat.decrementBy(harden({ Liquidity: liquidityAmountOut })),
+    );
+    if (feeSeat) {
+      zcf.reallocate(poolSeat, zcfSeat, feeSeat);
+    } else {
+      zcf.reallocate(poolSeat, zcfSeat);
+    }
+    zcfSeat.exit();
+    updateUpdaterState(updater, pool);
+    return 'Added liquidity.';
+  };
 
-        const tokenKeywordAmountOut = AmountMath.make(
-          secondaryBrand,
-          calcValueToRemove(
-            liqTokenSupply,
-            pool.getSecondaryAmount().value,
-            liquidityValueIn,
-          ),
-        );
+  const makePool = defineKind(
+    'pool',
+    (liquidityZcfMint, poolSeat, secondaryBrand) => {
+      const { brand: liquidityBrand, issuer: liquidityIssuer } =
+        liquidityZcfMint.getIssuerRecord();
+      const { notifier, updater } = makeNotifierKit();
 
-        liqTokenSupply -= liquidityValueIn;
+      return {
+        liqTokenSupply: 0n,
+        liquidityIssuer,
+        poolSeat,
+        liquidityBrand,
+        secondaryBrand,
+        liquidityZcfMint,
+        updater,
+        notifier,
+        vPool: undefined,
+        toCentralPriceAuthority: undefined,
+        fromCentralPriceAuthority: undefined,
+      };
+    },
+
+    state => {
+      const {
+        liquidityIssuer,
+        poolSeat,
+        liquidityBrand,
+        secondaryBrand,
+        updater,
+        notifier,
+      } = state;
+
+      /** @type {XYKPool} */
+      const pool = {
+        getLiquiditySupply: () => state.liqTokenSupply,
```

Comment:
why is `liqTokenSupply` access through `state`, but `liquidityIsuser`, `poolSeat`, `updater`, etc. (line 126) are extracted and accessed as top-level values?